packages/expo-image/android/src/main/java/expo/modules/image/thumbhash/ThumbhashDecoder.kt

```diff
@@ -0,0 +1,366 @@
+package expo.modules.image.thumbhash
+
+import android.graphics.Bitmap
+import android.graphics.Color
+
+// ThumbHash Java implementation (converted to kotlin) thanks to @evanw https://github.com/evanw/thumbhash
+object ThumbhashDecoder {
+  /**
+   * Encodes an RGBA image to a ThumbHash. RGB should not be premultiplied by A.
+   *
+   * @param w    The width of the input image. Must be ≤100px.
+   * @param h    The height of the input image. Must be ≤100px.
+   * @param rgba The pixels in the input image, row-by-row. Must have w*h*4 elements.
+   * @return The ThumbHash as a byte array.
+   */
+  fun rgbaToThumbHash(w: Int, h: Int, rgba: ByteArray): ByteArray {
+    // Encoding an image larger than 100x100 is slow with no benefit
+    require(!(w > 100 || h > 100)) { w.toString() + "x" + h + " doesn't fit in 100x100" }
+
+    // Determine the average color
+    var avg_r = 0f
+    var avg_g = 0f
+    var avg_b = 0f
+    var avg_a = 0f
+    run {
+      var i = 0
+      var j = 0
+      while (i < w * h) {
+        val alpha = (rgba[j + 3].toInt() and 255) / 255.0f
+        avg_r += alpha / 255.0f * (rgba[j].toInt() and 255)
+        avg_g += alpha / 255.0f * (rgba[j + 1].toInt() and 255)
+        avg_b += alpha / 255.0f * (rgba[j + 2].toInt() and 255)
+        avg_a += alpha
+        i++
+        j += 4
+      }
+    }
```

Comment:
```suggestion
    var i = 0
    var j = 0
    while (i < w * h) {
      val alpha = (rgba[j + 3].toInt() and 255) / 255.0f
      avg_r += alpha / 255.0f * (rgba[j].toInt() and 255)
      avg_g += alpha / 255.0f * (rgba[j + 1].toInt() and 255)
      avg_b += alpha / 255.0f * (rgba[j + 2].toInt() and 255)
      avg_a += alpha
      i++
      j += 4
    }
```
that run block doesn't seem to be needed.